src/components/structures/TimelinePanel.js

```diff
@@ -1023,7 +1020,16 @@ var TimelinePanel = React.createClass({
             var boundingRect = node.getBoundingClientRect();
             if ((allowPartial && boundingRect.top < wrapperRect.bottom) ||
                 (!allowPartial && boundingRect.bottom < wrapperRect.bottom)) {
-                return i;
+                // find last invisible event before the next visible one.
```

Comment:
Not sure if this is accurate. Or maybe I'm confused.

Surely `j` (which potentially `=== i`) is the index of the the first event that should be shown? So you're actually finding the event before the first visible event, which could still be `i`.